src/ephy-bookmark-properties-grid.c

```diff
@@ -376,8 +394,58 @@ ephy_bookmark_properties_grid_destroy (GtkWidget *widget)
 {
   EphyBookmarkPropertiesGrid *self = EPHY_BOOKMARK_PROPERTIES_GRID (widget);
 
+  if (ephy_bookmark_is_uploaded (self->bookmark) == FALSE)
+    goto out;
+
+  /* Check if any actual changes were made to the name, address or tags. If yes,
+   * set the uploaded flag to FALSE. */
+
+  if (self->prev_name != NULL) {
+    if (g_strcmp0 (self->prev_name, ephy_bookmark_get_title (self->bookmark)) != 0) {
```

Comment:
I prefer to write one conditional with and `&&` condition, rather than two nested conditionals. It doesn't matter much when there is so little nesting and two-space indentations, but it's probably better style and in more complex cases with more indentation it's much nicer.